ygnmi/ygnmi.go

```diff
@@ -141,13 +141,45 @@ func NewClient(c gpb.GNMIClient, opts ...ClientOption) (*Client, error) {
 	return yc, nil
 }
 
+// Option can be used modify the behavior of the gNMI requests.
+type Option func(*opt)
+
+type opt struct {
+	useGet bool
+	mode   gpb.SubscriptionMode
+}
+
+func resolveOpts(opts []Option) *opt {
+	o := &opt{}
+	for _, opt := range opts {
+		opt(o)
+	}
+	return o
+}
```

Comment:
It was a bit confusing reading this with all the different opts and their meanings. In this case I think it would be nice for the reader to see a doc comment that mentions how it's resolving these options.